litellm/utils.py

```diff
@@ -4249,6 +4249,7 @@ def get_llm_provider(
             model.split("/", 1)[0] in litellm.provider_list
             and model.split("/", 1)[0] not in litellm.model_list
             and len(model.split("/"))
+            and not api_key
```

Comment:
hey @Raymond1415926 this isn't what i thought you meant, when you said 'applied globally'

please add the check for each provider in the block

Reply:
reference implementation: https://github.com/BerriAI/litellm/blob/eaee90b02e6a0a9a20ddad5d9bf70136799511db/litellm/utils.py#L4304